store/src/java/com/zimbra/cs/mailbox/ContactAutoComplete.java

```diff
@@ -409,6 +409,20 @@ public AutoCompleteResult resolveEmailAddr(String str) throws ServiceException {
              }
            return result;
     }
+
+    public AutoCompleteResult query(String str) throws ServiceException {
+    	AutoCompleteResult result = new AutoCompleteResult(20);
+    	result.rankings = new ContactRankings(getRequestedAcctId());
+    	
+        Pair<List<Folder>, Map<ItemId, Mountpoint>> pFolders = getLocalRemoteContactFolders(null);
+        List<Folder> listFolders = pFolders.getFirst();
+        final String searchContactFolderQuery = generateFolderQuery(listFolders);
+        
+    	String query = searchContactFolderQuery + generateQuery(str.toLowerCase());
+        queryFolders(str.toLowerCase(), query, null, 20, result);
```

Comment:
What is 20 here? Any specific reason to keep it as 20

Reply:
20 is the number to limit the results, referred it from AutoComplete, but I think now it should be 1, as against one email Id we must be having one emailed contact stored into the DB. Rectified. Thanks